src/OpenColorIO/GpuShaderDesc.cpp

```diff
@@ -110,14 +125,7 @@ void GpuShaderCreator::setLanguage(GpuLanguage lang) noexcept
     AutoMutex lock(getImpl()->m_cacheIDMutex);
        
     getImpl()->m_language = lang;
-    if(lang == GPU_LANGUAGE_MSL_2_0)
-    {
-        getImpl()->m_classWrappingInterface = std::unique_ptr<MetalShaderClassWrapper>(new MetalShaderClassWrapper);
-    }
-    else
-    {
-        getImpl()->m_classWrappingInterface = std::unique_ptr<NullGpuShaderClassWrapper>(new NullGpuShaderClassWrapper);
-    }
+    getImpl()->updateClassWrappingInterface();
```

Comment:
So, the code is then
`getImpl()->m_classWrappingInterface = Impl::CreateClassWarpper(m_language);`

Reply:
@hodoulp Thanks for the suggestion. I applied the requested change.